v_4_3_0/files/k8s-resource/calico-all.yaml

```diff
@@ -351,7 +351,7 @@ spec:
         k8s-app: calico-kube-controllers
     spec:
       nodeSelector:
-        beta.kubernetes.io/os: linux
+        kubernetes.io/os: linux
```

Comment:
Just for transparency, could you put here the source?

is it needed for calico upgrade?

Reply:
I've found it in the deprecation list 

> OS and Arch information is now recorded in kubernetes.io/os and kubernetes.io/arch labels on Node objects. The previous labels (beta.kubernetes.io/os and beta.kubernetes.io/arch) are still recorded, but are deprecated and targeted for removal in v1.18. (#73333, @yujuhong)

Reply:
we use https://docs.projectcalico.org/v3.2/releases#v3.6.1 as source for calico manifests and than just update when they do

Reply:
I'm going to review the labels here because in KVM `kube-calico-controllers` does not go up due to some labels misalignment 

```
Events:
  Type     Reason            Age                  From               Message
  ----     ------            ----                 ----               -------
  Warning  FailedScheduling  83s (x16 over 6m6s)  default-scheduler  0/4 nodes are available: 4 node(s) didn't match node selector.
```

Reply:
yeah, i thnik Tuomas recently introduced label node selector so it sits on master

Reply:
I've personally verified the upstream calico manifests for the version 3.6.1 and no major changes here.

Reply:
> yeah, i thnik Tuomas recently introduced label node selector so it sits on master

Yeah and that was one of the things that had to be refined as we dropped the old `role=master` label from node. That might cause the problem here?